pkg/controller/machine/openshift_controller_test.go

```diff
@@ -0,0 +1,37 @@
+package machine
+
+import (
+	"errors"
+	controllerError "github.com/openshift/cluster-api/pkg/controller/error"
+	"testing"
+	"time"
+)
+
+func mockDrainNode(shouldDelay bool) error {
+	if shouldDelay {
+		return &controllerError.RequeueAfterError{RequeueAfter: 20 * time.Second}
+	}
+	return errors.New("other error")
+}
+
+func TestDelayAfterDrainFail(t *testing.T) {
+	tCases := []struct {
+		shouldDelay bool
+		expectedErr bool
+	}{
+		{
+			shouldDelay: true,
+			expectedErr: false,
+		},
+		{
+			shouldDelay: false,
+			expectedErr: true,
+		},
+	}
+	for i, tc := range tCases {
+		_, err := delayAfterDrainFail(mockDrainNode(tc.shouldDelay))
+		if err != nil && tc.expectedErr == false {
```

Comment:
Where do we assert for `exptectedErr == true`?

Reply:
It's implied via the err != nil condition.

It's a little confusing, I'll grant you that.  delayAfterDrainFail either returns a struct and nil, or an empty struct and err.  Essentially, there are only two valid cases, and we test them both.